client/src/components/Download.js

```diff
@@ -240,26 +68,53 @@ export const Download = ({ Icon, computedFile: publicComputedFile }) => {
       }
     }
 
-    if (!download && token && showing) asyncFetch()
-  }, [download, token, showing])
+    if (!download && token && showing && publicComputedFile) asyncFetch()
+  }, [download, token, showing, publicComputedFile])
+
   return (
     <>
-      {Icon ? (
-        <Anchor icon={Icon} onClick={handleClick} />
+      {icon ? (
+        <Anchor icon={icon} onClick={handleClick} />
       ) : (
         <Button
+          aria-label={label}
           flex="grow"
           primary
           label={label}
-          aria-label={label}
-          disabled={!publicComputedFile}
+          disabled={publicComputedFile === undefined}
           onClick={handleClick}
         />
       )}
-      <Modal showing={showing} setShowing={setShowing} title={label}>
-        <Box width={{ width: 'full' }}>
-          {download ? <DownloadView computedFile={download} /> : <TokenView />}
-        </Box>
+      <Modal title={label} showing={showing} setShowing={setShowing}>
+        {publicComputedFile && (
+          <Modal.Header>
```

Comment:
can you export this as `ModalHeader` instead of nesting it inside the other component